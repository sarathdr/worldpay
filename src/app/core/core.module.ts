import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
