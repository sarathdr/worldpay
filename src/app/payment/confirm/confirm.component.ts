import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Cart } from '../../core/models/cart.interface';

@Component({
  selector: 'payment-confirm',
  template: `
    <div class="content">
      <h1>Your payment details has been authorized</h1>
      <p>Confirm the payment to transfer money</p>
      <p><b>Total</b>{{ cart.total | currency: cart.currency }}</p>
      <div class="row">
        <button mat-raised-button type="button" (click)="cancelPayment.emit()">Cancel</button>
        <button mat-raised-button color="primary" type="button" (click)="confirmPayment.emit()">Confirm</button>
      </div>
    </div>
  `,
  styles: [
      `
          .content {
              display: flex;
              flex-direction: column;
              justify-content: space-between;
              width: 400px;
              height: 285px;
              margin: 0 auto;
          }

          .row {
              display: flex;
              flex-direction: row;
              justify-content: space-between;
          }
    `
  ]
})
export class ConfirmComponent {
  @Input() cart: Cart;

  @Output() confirmPayment = new EventEmitter<void>();
  @Output() cancelPayment = new EventEmitter<void>();
}
