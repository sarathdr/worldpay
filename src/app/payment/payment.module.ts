import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthorizeComponent } from './authorize/authorize.component';
import { AuthorizeContainer } from './authorize/authorize.container';
import { ConfirmComponent } from './confirm/confirm.component';
import { ConfirmContainer } from './confirm/confirm.container';
import { RefundComponent } from './refund/refund.component';
import { RefundContainer } from './refund/refund.container';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { MatButtonModule, MatFormFieldModule, MatInputModule, MatSelectModule } from '@angular/material';
import { AuthorizeFormComponent } from './authorize/authorize-form.component';
import { ConfirmPaymentGuard } from './guards/confirm-payment.guard';
import { ExpiryDateModule } from '../core/directives/expiry-date.module';
import { RefundPaymentGuard } from './guards/refund-payment.guard';
import { RefundSuccessComponent } from './refund/refund-success.component';

@NgModule({
  declarations: [
    AuthorizeComponent,
    AuthorizeFormComponent,
    AuthorizeContainer,
    ConfirmComponent,
    ConfirmContainer,
    RefundComponent,
    RefundSuccessComponent,
    RefundContainer
  ],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule.forChild([
      {
        path: '',
        component: AuthorizeContainer
      },
      {
        path: 'confirm',
        component: ConfirmContainer,
        canActivate:[ConfirmPaymentGuard],
        canDeactivate:[ConfirmPaymentGuard]
      },
      {
        path: 'refund',
        component: RefundContainer,
        canActivate:[RefundPaymentGuard],
        canDeactivate:[RefundPaymentGuard]
      }
    ]),
    MatFormFieldModule,
    MatInputModule,
    MatSelectModule,
    ReactiveFormsModule,
    MatButtonModule,
    ExpiryDateModule
  ],
  providers:[
    ConfirmPaymentGuard,
    RefundPaymentGuard
  ]
})
export class PaymentModule { }
