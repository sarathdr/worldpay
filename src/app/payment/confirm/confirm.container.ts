import { Component, OnDestroy } from '@angular/core';
import { CartService } from '../../core/services/cart.service';
import { PaymentService } from '../../core/services/payment.service';
import { Router } from '@angular/router';
import { Observable, Subscription } from 'rxjs';
import { Cart } from '../../core/models/cart.interface';
import { take } from 'rxjs/operators';

@Component({
  selector: 'payment-confirm-page',
  template: `
    <payment-confirm
        [cart]="cart$ | async"
        (cancelPayment)="cancelPayment()"
        (confirmPayment)="confirmPayment()">
    </payment-confirm>
  `
})
export class ConfirmContainer {
  cart$: Observable<Cart>;

  constructor(private cartService: CartService,
              private paymentService: PaymentService,
              private router: Router) {
    this.cart$ = cartService.getCart();
  }

  confirmPayment() {
    this.paymentService.confirmPayment()
      .pipe(take(1))
      .subscribe(() => this.router.navigate(['payment', 'refund']));
  }

  cancelPayment() {
    this.paymentService.cancelPayment()
      .pipe(take(1))
      .subscribe(() => this.router.navigate(['home']));
  }
}
