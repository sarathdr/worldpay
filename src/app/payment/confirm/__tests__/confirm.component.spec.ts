import { TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { ConfirmComponent } from '../confirm.component';
import { Currency } from '../../../core/models/payment.interface';

describe('ConfirmComponent', () => {

  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MatListModule,
        MatButtonModule,
        MatTableModule
      ],
      declarations: [
        ConfirmComponent,
        TestConfirmComponent
      ],
    });

    fixture = TestBed.createComponent(TestConfirmComponent);
  });

  it('should render the component correctly', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  @Component({
    selector: 'test-payment-confirm ',
    template: `
      <payment-confirm [cart]="cart"></payment-confirm>`
  })
  class TestConfirmComponent {
    cart = {
      products: [{
        name: 'Pen',
        price: 10
      },
        {
          name: 'Pencil',
          price: 12
        }],
      total: 22,
      currency: Currency.GBP
    };
  }

});
