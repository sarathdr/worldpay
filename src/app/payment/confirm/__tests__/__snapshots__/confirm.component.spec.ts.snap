// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmComponent should render the component correctly 1`] = `
<test-payment-confirm
  cart={[Function Object]}
>
  <payment-confirm
    ng-reflect-cart="[object Object]"
  >
    <div
      class="content"
    >
      <h1>
        Your payment details has been authorized
      </h1>
      <p>
        Confirm the payment to transfer money
      </p>
      <p>
        <b>
          Total
        </b>
        Â£22.00
      </p>
      <div
        class="row"
      >
        <button
          class="mat-raised-button mat-button-base"
          mat-raised-button=""
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            Cancel
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <button
          class="mat-raised-button mat-button-base mat-primary"
          color="primary"
          mat-raised-button=""
          ng-reflect-color="primary"
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            Confirm
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </div>
    </div>
  </payment-confirm>
</test-payment-confirm>
`;
